### ISSUE TITLE
Hook up with continuous integration.

### ISSUE THREAD

**Issue by codebycliff** [Iowa-Elixir/reunions#8]
I'm going to try CircleCI first. That seems to be the majority of resources out there when searching "Elixir Continuous Integration".

I might also try SemaphoreCI, as it claims to have zero config for elixir

**Comment by joseluistorres** [comment]
Sounds good to me!

**Comment by ctreptow** [comment]
We use Travis, and their Elixir looks pretty similar to what we do:

https://docs.travis-ci.com/user/languages/elixir/

However, zero config is better if it works!

**Comment by ctreptow** [comment]
I noticed that both CircleCI and SemaphoreCI ran on @cartothemax 's [CRUD PR](https://github.com/Iowa-Elixir/reunions/pull/16).  Was that by design right now?

**Comment by cartothemax** [comment]
@ctreptow the errors coming back are due to no `ENV` variables being set.  When the CI tools trying to run `mix test` it fails when it doesn't have valid credentials for creating a test database.  

That being said going forward having both CircleCI and SemaphoreCI run automatically would be best in my opinion.

**Comment by codebycliff** [comment]
As soon as I connected both the continuous integration systems to Github, it set up the hooks. The build won't pass until we get the configuration into master for whichever one we decide. As far as which one to use, I don't have a strong opinion. We use CircleCI at work, so I'm already in there. However, semaphore seems to have better support for Elixir/Phoenix. I'm leaning toward Semaphore for that reason. Here are the two branches I have:

[CircleCI](https://github.com/Iowa-Elixir/reunions/compare/continuous-integration/circleci?expand=1)

[Semaphore](https://github.com/Iowa-Elixir/reunions/compare/continuous-integration/semaphore?expand=1)

Let me know what you guys think and we can move forward with one of them.

**Comment by cartothemax** [comment]
Semaphore is my vote!  I like the name more and appreciate it supports Elixir out of the box.

**Comment by chuck-john** [comment]
Semaphore because it's new to me.  Although, I'm curious if there are any performance differences between the two.  Is it possible to use both for a few PRs before making a final decision?

**Comment by codebycliff** [comment]
Cool. I'll open the PR then.

**Comment by joseluistorres** [comment]
Yes I'd like to see how they both work with Elixir/Phoenix.

**Comment by codebycliff** [comment]
Definitely. We'll just have to merge both pull requests. For what it's worth, I believe CircleCI is noticeably slower than Semaphore with this particular setup.

**Comment by codebycliff** [comment]
Both pull requests are up. I've tried adding you all to Semaphore, but it's not very intuitive and doesn't go off of Github permissions as you would suspect. Let me know if you don't have access, and I'll try again. CircleCI does go off of Github permissions, so everyone should have access to that.

**Comment by ctreptow** [comment]
Just discovered Credo.  This might be useful for us at this stage: http://credo-ci.org/

**Comment by joseluistorres** [comment]
@ctreptow yes! I think @codebycliff already added it https://github.com/Iowa-Elixir/reunions/blob/master/mix.exs#L41
